src/java.base/share/classes/com/sun/crypto/provider/PBKDF2KeyImpl.java

```diff
@@ -122,7 +123,7 @@ private static byte[] getPasswordBytes(char[] passwd) {
             }
         }
         // Use the cleaner to zero the key when no longer referenced
-        final byte[] k = this.key;
+        final byte[] k = this.key = key;
         final char[] p = this.passwd;
```

Comment:
Now that `this.key` and `this.passwd` is final, we can get rid of the local final variables `k` and `p`, right?